### ISSUE TITLE
Gcc/Intel+Pthreads with Hwloc Numerical Exception Thrown in unit tests (SKX arch): KokkosCore_UnitTest_Threads, KokkosCore_UnitTest_Default, KokkosCore_PerfTestExec

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#3460]
These tests began failing with `***Exception: Numerical` in nightlies 10/06/2020, errors occur with these specific tests (reproduced on Blake):

```bash
threads.policy_construction
defaultdevicetype.reduce_instantiation_a3
default_exec.overlap_team_policy
```

Sample reproducer on Blake testbed with intel/19 (also occurs with gcc/7.2.0):
```bash
module load cmake/3.12.3 intel/compilers/19.1.144 gcc/7.2.0 numa/2.0.11 hwloc/1.11.8

export OMP_PROC_BIND=spread
export OMP_PLACES=threads

../generate_makefile.bash \
    --arch=SKX \
    --cxxflags="-Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized" \
    --with-hwloc=$HWLOC_ROOT \
    --compiler=icpc \
    --with-devices=Pthread
```

(Note some of the cxxflags are unsupported by intel, just copy/pasted here to avoid altering reproducer)

**Comment by ndellingwood** [comment]
Went through bisection on merge commits, looks like the failures began with merge of PR #3206, @DavidPoliakoff are you able to look into this?

Failing SHA: 153d255355c1465f39b1c0bd72f282bad29d277c
Passing SHA: 5c9f4a62e6dfa2a5729bd5e4739d3d24f15d7859

**Comment by ndellingwood** [comment]
Should be addressed by #3510 , will confirm nightlies pass this evening
